### PR TITLE
lnav: fix scroll issue by depending on ncurses

### PR DESCRIPTION
**Pull request by ian-h-chamberlain**
Fixes https://github.com/tstack/lnav/issues/1190

Depend on, include and link homebrew version of `ncurses`. This more closely matches the upstream build configuration:

https://github.com/tstack/lnav/blob/master/.github/workflows/bins.yml#L135-L144

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

cc @tstack

### CODE REVIEW

File: Formula/l/lnav.rb [hunk 1]
```diff
@@ -41,11 +43,15 @@ class Lnav < Formula
   def install
     system "./autogen.sh" if build.head?
     ENV.append "LDFLAGS", "-L#{Formula["libarchive"].opt_lib}"
+    ENV.append "LDFLAGS", "-L#{Formula["ncurses"].opt_lib}"
+    ENV.append "CXXFLAGS", "-I#{Formula["ncurses"].opt_include}"
```
Comment (SMillerDev) [inline Formula/l/lnav.rb:47]: Can autogen not detect these?

Comment (ian-h-chamberlain) [reply Formula/l/lnav.rb:47]: I'm not sure, maybe a question for @tstack if they recall why this had to be plumbed in for `libarchive` (5194fe31057ab6e100c6db3006856b308321fd2b). If you mean literally `autogen.sh` — it's only run if `build.head?`, so it wouldn't apply for regular installs, I think.

I can try installing without these and see if it picks up the right version automatically.

Comment (tstack) [reply Formula/l/lnav.rb:47]: There's a `--with-ncurses` flag you can pass to configure.  But, it's not much different from adding the path to the `LDFLAGS` and `CXXFLAGS`.

Comment (SMillerDev) [reply Formula/l/lnav.rb:47]: It is much easier to read for maintainers though, so let's use that

Comment (ian-h-chamberlain) [reply Formula/l/lnav.rb:47]: Sounds good. In that case is `libarchive` still needed in `LDFLAGS` too, or should I also remove that while I'm in there?

Comment (ZhongRuoyu) [reply Formula/l/lnav.rb:47]: If it works without the explicit `LDFLAGS` (which I presume it should, given the `--with-libarchive` flag below): yes, let's remove that for the same reason.

**Review by ZhongRuoyu — APPROVED** [review]
Thank you @ian-h-chamberlain.

**Pull request merged**